server/pbench/bin/pbench-verify-backup-tarballs

```diff
@@ -91,8 +94,61 @@ if cd $backup ;then
     done < $controllers | sort > $listb
 fi
 
+# make sure the directories exists
+mkdir -p $backup/QUARANTINE
+mkdir -p $primary/QUARANTINE
+
+# move the tarballs in backup, to quarantine directory, that fail the MD5 check
+# for now backup supposed to be in the same server as archive
+grep FAIL $listb > $listb.failed
+cat $listb.failed | sed -e "s/: FAILED$//" > $quarantine.list
+
+IFS=$'\n'
+for i in `cat $quarantine.list`;do
+    if [ -f $i ];then
+        mv $i* QUARANTINE
+    fi
+done
+
 # construct the report
 > $report
+if [ -s $quarantine.list ] ;then
+    (echo "* In $backup: The calculated MD5 of the following entries failed to match the stored MD5, 
+    so moving to quarantine directory"; cat $quarantine.list | sed 's;^\.;'$backup';'; echo) >> $report
+fi
+
+# restore the tarballs in archive from backup, that fail the MD5 check
+grep FAIL $listp > $listp.failed
+cat $listp.failed | sed -e "s/: FAILED$//" > $restore.list
+grep OK $listb > $listb.ok
+cat $listb.ok | sed -e "s/: OK$//" > $backuplist.ok
+
+(echo "* In $primary: The calculated MD5 of the following entries failed to match the stored MD5,
+so trying to restore from backup, otherwise moving to quarantine"; cat $quarantine.list | sed 's;^\.;'$primary';';) >> $report
+
+IFS=$'\n'
+for i in `cat $restore.list`;do
+    if [ -f $i ];then
+        grep -q $i $backuplist.ok
+        if [ $? -eq 0 ];then
+            archivedir=`echo ${i%/*} | sed 's;^\.;'$primary';'`
+            cp $i* $archivedir
```

Comment:
That catches the tarball and its MD5 sum, but probably not any associated prefix file.